Tests/ExtDataRoot_GridComp.F90

```diff
@@ -509,40 +554,63 @@ subroutine FillState(inState,outState,time,grid,Synth,rc)
       integer, allocatable :: seeds(:)
       type(ESMF_VM) :: vm
 
-      call MAPL_GridGet(grid,localcellcountperdim=ldims,_RC)
-      call MAPL_Grid_Interior(grid,i1,in,j1,jn)
+      if (synth%on_tiles) then
+
+      else
+         call MAPL_GridGet(grid,localcellcountperdim=ldims,_RC)
+         call MAPL_Grid_Interior(grid,i1,in,j1,jn)
+      end if
       call ESMF_StateGet(outState,itemcount=itemCount,_RC)
       allocate(outNameList(itemCount),stat=status)
       _VERIFY(status)
       call ESMF_StateGet(outState,itemNameList=outNameList,_RC)
 
-      call MAPL_GetPointer(inState,exPtr2,'time',_RC)
-      exPtr2=synth%tFunc%evaluate_time(Time,_RC)
+      if (synth%on_tiles) then
+         call MAPL_GetPointer(inState,exPtr1,'time',_RC)
+         exPtr1=synth%tFunc%evaluate_time(Time,_RC)
+      else
+         call MAPL_GetPointer(inState,exPtr2,'time',_RC)
+         exPtr2=synth%tFunc%evaluate_time(Time,_RC)
+      end if
 
-      call MAPL_GetPointer(inState,exPtr2,'i_index',_RC)
-      do j = 1,ldims(2)
-         do i=1,ldims(1)
-            exPtr2(i,j)=i1+i-1
+      if (synth%on_tiles) then
+      
+      else
+         call MAPL_GetPointer(inState,exPtr2,'i_index',_RC)
+         do j = 1,ldims(2)
+            do i=1,ldims(1)
+               exPtr2(i,j)=i1+i-1
+            enddo
          enddo
-      enddo
-      call MAPL_GetPointer(inState,exPtr2,'j_index',_RC)
-      do i = 1,ldims(1)
-         do j=1,ldims(2)
-            exPtr2(i,j)=j1+j-1
+         call MAPL_GetPointer(inState,exPtr2,'j_index',_RC)
+         do i = 1,ldims(1)
+            do j=1,ldims(2)
+               exPtr2(i,j)=j1+j-1
+            enddo
          enddo
-      enddo
+      end if
 
-      call MAPL_GetPointer(inState,exPtr2,'doy',_RC)
-      exPtr2 = compute_doy(time,_RC)
+      if (synth%on_tiles) then
+         call MAPL_GetPointer(inState,exPtr1,'doy',_RC)
+         exPtr1 = compute_doy(time,_RC)
+      else
+         call MAPL_GetPointer(inState,exPtr2,'doy',_RC)
+         exPtr2 = compute_doy(time,_RC)
+      end if
 
-      call MAPL_GetPointer(inState,exPtr2,'rand',_RC)
       call random_seed(size=seed_size)
       allocate(seeds(seed_size))
       call ESMF_VMGetCurrent(vm,_RC)
       call ESMF_VMGet(vm,localPet=mypet,_RC)
       seeds = mypet
       call random_seed(put=seeds)
-      call random_number(exPtr2)
+      if (synth%on_tiles) then
+         call MAPL_GetPointer(inState,exPtr1,'rand',_RC)
+         call random_number(exPtr1)
+      else
+         call MAPL_GetPointer(inState,exPtr2,'rand',_RC)
+         call random_number(exPtr2)
+      end if
```

Comment:
ok - I'm curious.  What does randomness have to do with this?

Reply:
@tclune, a random number of one of the items in the "menu" of things you can use to fill exports, so that's what is has to do with it